device/transport/amqp/devdoc/amqp_twin_client_requirements.md

```diff
@@ -3,106 +3,139 @@
 
-## Public API
+**SRS_NODE_DEVICE_AMQP_TWIN_06_005: [** The `AmqpDeviceMethodClient` shall inherit from the `EventEmitter` class. **]**
 
-### Constructor
+### getTwin(callback: (err: Error, twin?: TwinProperties) => void): void;
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_004: [** The `AmqpTwinClient` constructor shall throw `ReferenceError` if the `config` object is falsy. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_007: [** The `getTwin` method shall attach the sender link if it's not already attached. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_002: [** The `AmqpTwinClient` constructor shall throw `ReferenceError` if the `client` object is falsy. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_008: [** If attaching the sender link fails, the `getTwin` method shall call its callback with the error that caused the failure. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_005: [** The `AmqpDeviceMethodClient` shall inherit from the `EventEmitter` class. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_009: [** THe `getTwin` method shall attach the receiver link if it's not already attached. **]**
 
-### response event
+**SRS_NODE_DEVICE_AMQP_TWIN_16_010: [** If attaching the receiver link fails, the `getTwin` method shall call its callback with the error that caused the failure. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_006: [** When a listener is added for the `response` event, and the `post` event is NOT already subscribed, upstream and downstream links are established via calls to `attachReceiverLink` and `attachSenderLink`. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_011: [** The `getTwin` method shall send an `AmqpMessage` using the `SenderLink.send` method with the following annotations and properties:
+- `operation` annotation set to `GET`.
+- `resource` annotation set to `undefined`
+- `correlationId` property set to a random integer
+- `body` set to ` `. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_007: [** The endpoint argument for attacheReceiverLink shall be `/devices/<deviceId>/twin`. **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_012: [** If the `SenderLink.send` call fails the `getTwin` method shall call its callback with the error that caused the failure. **]**
 
-**SRS_NODE_DEVICE_AMQP_TWIN_06_008: [** The link options argument for attachReceiverLink shall be:
- attach: {
-        properties: {
-          'com.microsoft:channel-correlation-id' : 'twin:<correlationId>',
-          'com.microsoft:api-version' : endpoint.apiVersion
-        },
-        sndSettleMode: 1,
-        rcvSettleMode: 0
-      } **]**
+**SRS_NODE_DEVICE_AMQP_TWIN_16_013: [** The `getTwin` method shall monitor `Message` objects on the `ReceiverLink.on('message')` handler and until a message with the same `correlationId` as the one that was sent is received. **]**
 
+**SRS_NODE_DEVICE_AMQP_TWIN_16_014: [** The `getTwin` method shall parse the body of the received message and call its callback with a `null` error object and the parsed object as a result. **]**
```

Comment:
>call its callback with a `null` error object  [](start = 110, length = 45)

I think this is what you were referring to when you mentioned twin code not checking the status code, but, if not, it looks like this code doesn't check the status code :)

Reply:
yeah and since the annotations are only surfaced through the transportObj i'm not sure how to do this cleanly...

---
In reply to: [171683343](https://github.com/pierreca/azure-iot-sdk-node/pull/1#discussion_r171683343) [](ancestors = 171683343)

Reply:
I've bugged it and will address this separately

---
In reply to: [171741000](https://github.com/pierreca/azure-iot-sdk-node/pull/1#discussion_r171741000) [](ancestors = 171741000,171683343)